### PR TITLE
Added third argument to specify publishing frequency

### PR DESCRIPTION
**Pull request by adnanmunawar**
Hello, 

I myself needed this change and I have read several posts online for people requesting this change. I have added room for providing the third argument to the executable that specifies the publishing frequency, which was hardcoded to 1 sec before. As can be seen from the code, the third argument can even be not provided and the code will use a default publishing frequency of 500 Hz

### CODE REVIEW

**Comment by tfoote** [comment]
Can you explain a little bit more about your use case? Making the rate adjustable is a reasonable enhancement. A default of 500 Hz for a command like tool is very fast.

**Comment by tfoote** [comment]
@ros-pull-request-builder

**Comment by adnanmunawar** [comment]
Sure, so all I did was that I added an optional third argument for input to the
rosrun tf tf_echo required:source_frame required:target_frame optional:publishing_frequency

This 3rd argument is optional and thus if the user does not provided one, the rate will remain fixed at 500 Hz. As far as the rate is concerned, sure, I shall make it to 1 second as before, thus if the use does not provide new publishing rate, a default rate of 1 sec is used

**Comment by tfoote** [comment]
This looks reasonable. Can you explain why you want it outputting at 500Hz. Most terminals are going to have a very hard time keeping up with that data rate.

**Comment by adnanmunawar** [comment]
I understand that 500 Hz is too fast for terminals, specially on mobile robots. Hence I modified the code to keep the default publishing at 1 Hz unless you want a faster publishing rate, in which case you specify the rate as the third argument

**Comment by tfoote** [comment]
Yes, I generally agree that adding this capability is good however please
explain your use case, I'd like to understand it more generally. I'm
guessing there's a better solution than making tf_echo run at 500 Hz. And
if you have a use case others might have the same use case. I'd like to
make sure that this is the best approach before committing to an API
change.

**Comment by adnanmunawar** [comment]
Sure, so I would provided one reference to support my issue, while google-ing up for a solution I came across this ticket https://code.ros.org/trac/ros-pkg/ticket/5480 
I am sure I can find some more tickets/discussions regarding the same issue. In my case, I am using a medical robot, the daVinci Surgical Robot. It's base links move pretty fast for a small end effector movement and I use tf_echo as a debugging tool to get the frames of the base links very oftenly. In most cases, having the default publishing rate of 1Hz is very slow. I miss quite a lot of the movement of the base links due to slower publishing rate. At many a times, the base links reach a position and then come back the opposite way in less than half a second. I could write a new node to cater for it, but what the use if I already have debugging tool like tf_echo, and I could myself add a publishing rate to it, I will save quite a lot of time. I am sure this issue can be extrapolated to several robots.

File: tf/src/tf_echo.cpp [hunk 1]
```diff
@@ -66,8 +67,14 @@ int main(int argc, char ** argv)
     printf("Note: This is the transform to get data from target_frame into the source_frame.\n");
     return -1;
   }
+  if (argc = 4)
```
Comment (tfoote) [inline tf/src/tf_echo.cpp]: This should be == as well.

**Comment by tfoote** [comment]
I noticed one small issue. Could you fix that and collapse this into one commit then I think it should be good to go.

**Comment by adnanmunawar** [comment]
Good catch. I shall fix it right away

**Comment by adnanmunawar** [comment]
Hello,
I have merged the commits into one.
-AM

**Comment by tfoote** [comment]
Same as other. Thanks, this is close, however it does not catch and report usage for too many arguments. The <3 should also check for >4

**Pull request merged**